classes/observers/listeners/AssignEditors.php

```diff
@@ -0,0 +1,112 @@
+use PKP\context\SubEditorsDAO;
+use PKP\db\DAORegistry;
+use PKP\log\SubmissionEmailLogDAO;
+use PKP\log\SubmissionEmailLogEntry;
+use PKP\mail\mailables\SubmissionNeedsEditor;
+use PKP\notification\NotificationSubscriptionSettingsDAO;
+use PKP\observers\events\SubmissionSubmitted;
+use PKP\security\Role;
+
+class AssignEditors
+{
+    public function subscribe(Dispatcher $events): void
+    {
+        $events->listen(
+            SubmissionSubmitted::class,
+            AssignEditors::class
+        );
+    }
+
+    public function handle(SubmissionSubmitted $event)
+    {
+        /** @var SubEditorsDAO $subEditorsDao */
+        $subEditorsDao = DAORegistry::getDAO('SubEditorsDAO');
+        $assignedUserIds = $subEditorsDao->assignEditors($event->submission, $event->context);
+
+        if ($assignedUserIds->count()) {
+            return;
+        }
+
+        $managers = Repo::user()
+            ->getCollector()
+            ->filterByRoleIds([Role::ROLE_ID_MANAGER])
+            ->filterByContextIds([$event->context->getId()])
+            ->getMany();
+
+        if (!$managers->count()) {
+            return;
+        }
+
+        $notificationManager = new NotificationManager();
+        foreach ($managers as $manager) {
+            $notificationManager->createNotification(
+                Application::get()->getRequest(),
+                $manager->getId(),
+                Notification::NOTIFICATION_TYPE_EDITOR_ASSIGNMENT_REQUIRED,
+                $event->context->getId(),
+                Application::ASSOC_TYPE_SUBMISSION,
+                $event->submission->getId(),
+                Notification::NOTIFICATION_LEVEL_TASK
+            );
+        }
+
+        /** @var NotificationSubscriptionSettingsDAO $notificationSubscriptionSettingsDao */
+        $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
+        $emailTemplate = Repo::emailTemplate()->getByKey($event->context->getId(), SubmissionNeedsEditor::getEmailTemplateKey());
+        $mailable = new SubmissionNeedsEditor($event->context, $event->submission);
+        $mailable
+            ->from($event->context->getData('contactEmail'), $event->context->getData('contactName'))
+            ->subject($emailTemplate->getLocalizedData('subject'))
+            ->body($emailTemplate->getLocalizedData('body'));
+
+        foreach ($managers as $manager) {
+            $unsubscribed = in_array(
+                Notification::NOTIFICATION_TYPE_SUBMISSION_SUBMITTED,
+                $notificationSubscriptionSettingsDao->getNotificationSubscriptionSettings(
+                    NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY,
+                    $manager->getId(),
+                    $event->context->getId()
+                )
+            );
+
+            if ($unsubscribed) {
+                continue;
+            }
+
+            Mail::send($mailable->recipients([$manager]));
```

Comment:
@NateWr, there were 2 associated notification emails before, one related to `NOTIFICATION_TYPE_EDITOR_ASSIGNMENT_REQUIRED` (A new article has been submitted to which an editor needs to be assigned) and another one `NOTIFICATION_TYPE_SUBMISSION_SUBMITTED` (A new article, "Title," has been submitted). Does this mean that you combined them? They both are still present in profile notifications settings

Reply:
I believe that `NOTIFICATION_TYPE_EDITOR_ASSIGNMENT_REQUIRED` is only triggered if no editors are automatically assigned.